test/functional/provider/admin/messages/inbox_controller_test.rb

```diff
@@ -48,4 +48,24 @@ def test_index_system_message
     assert_select 'title', "Inbox - Index | Red Hat 3scale API Management"
     assert_select '#export-to-csv', false, 'Export all Messages'
   end
+
+  test 'Prevent sending reply without subject' do
```

Comment:
This is the `Provider::Admin::Messages::InboxControllerTest` but we're not calling any of its actions on those 2 new tests. For those two, we should call `post :reply` and assert that things work as expected.

Reply:
We should also have a test in case there's no body.

Reply:
Agreed to first comment can add test which will call post :reply but this test also do similar check as we drain the message and assert on the latest one

For the second comment we dont have any sort of validation on body we allow both nil and with text also so what sense does it make to add a separate test for it

Reply:
> we dont have any sort of validation on body we allow both nil and with text also so what sense does it make to add a separate test for it

From my understanding of the linked ticket, we should not allow emails to be sent without a subject or without a body.
If that's not the case, we should change the ticket and this PR's title and description.

Reply:
No we had a word with the team and then only we decided only to keep subject mandatory and not body because there are lot of  hardcoded Messages in the code with no body and so they just want subject to be the required filed

here is the link to the JIRA you can read comments for the calrity(https://issues.redhat.com/browse/THREESCALE-8393)